lib/modules/apostrophe-palette-global/index.js

```diff
@@ -129,13 +127,53 @@ module.exports = {
       {
         name: 'palette',
         label: 'Palette Fields',
-        fields: fieldNames
+        fields: fieldNames.concat(['paletteCounter'])
       }
     ]);
 
+    // This route serves the existing palette stylesheet, constructed from the global object
+    // We do it this way so the browser can cache the styles as often as possible
+    self.route('get', 'palette-styles', function (req, res) {
+      var paletteFields = self.apos.modules['apostrophe-global'].options.paletteFields;
+      var rules = [];
+
+      paletteFields.forEach(function(field) {
```

Comment:
This business probably takes some time, so while caching per-user is good, caching per-process would be better. How about caching the string in `self.latestStyles` and the paletteCounter in `self.latestCounter`, and comparing, and reusing the string if it's the same version that was requested.

Note that here we get to break the "don't attach data to `self`" rule because we are intentionally sharing it between requests.

Because of the counter we don't even have to worry about a cache lifetime.

(We could share it between processes but that's overkill because it only takes four or so requests for all four processes to build a copy anyway, and the overhead of asking mongodb for something from a cache would be higher.)

You could wrap this in a quick Date.now() thing first though, to see if there's actually any time involved!